test/edu/washington/escience/myria/operator/StreamingAggTest.java

```diff
@@ -0,0 +1,1113 @@
+package edu.washington.escience.myria.operator;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import org.joda.time.DateTime;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.operator.agg.MultiGroupByAggregate;
+import edu.washington.escience.myria.operator.agg.PrimitiveAggregator.AggregationOp;
+import edu.washington.escience.myria.operator.agg.SingleColumnAggregatorFactory;
+import edu.washington.escience.myria.operator.agg.SingleGroupByAggregate;
+import edu.washington.escience.myria.operator.agg.StreamingAggregate;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.storage.TupleBatchBuffer;
+
+/**
+ * Test cases for {@link StreamingAggregate} class. Source tuples are generated in sorted order on group keys, if any.
+ * Some of the tests are taken from those for {@link SingleGroupByAggregate} and {@link MultiGroupByAggregate} since
+ * StreamingAggregate is expected to behave the same way they do if input is sorted.
+ */
+public class StreamingAggTest {
+
+  /**
+   * Construct a TupleBatchBuffer to be used as source of aggregate. Fixed schema and sorted on grouping columns.
+   *
+   * @param numTuples number of tuples to be added
+   * @return filled TupleBatchBuffer with each group key having (numTuples/10) tuples
+   */
+  private TupleBatchBuffer fillInputTbb(final int numTuples) {
+    final Schema schema =
+        new Schema(ImmutableList.of(Type.INT_TYPE, Type.DOUBLE_TYPE, Type.FLOAT_TYPE, Type.LONG_TYPE,
```

Comment:
Readability might be improved by using the `Schema.ofFields(name1, type1, name2, type2, ...)` factory method.